### PR TITLE
feat: Add nat-bus option to generate compose files for NATS

### PR DESCRIPTION
**Pull request by lenny-intel**
Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [x] I have opened a PR for the related docs change (if not, why?)
  https://github.com/edgexfoundry/edgex-docs/pull/846

## Testing Instructions
run `make build` from Compose Builder using this branch
Verify the none of the generated compose files changed.
Build ASC, edge-go services, Device virtual docker images with NATS. (add `-tags "include_nats_messaging` to each build target)
run `make gen nats-bus no-secty dev app-dev`
Modify compose file to use dev image of device-virtual
Verify the generated compose file has the proper overrides for NATS
Run `make up` 
Verify all service are connected to NAT Sever w/o error
Change app-rules-engine loglevel to DEBUG
Verify event are flowing
_

> Note eKuiper will not have NATS support yet and will fail to connect

_

### CODE REVIEW

**Review by AlexCuse — APPROVED** [review]

**Review by jim-wang-intel — DISMISSED** [review]
LGTM

**Comment by lenny-intel** [comment]
recheck

**Comment by lenny-intel** [comment]
@cherrycl , can you look at why this PR is failing the Smoke Tests?
I doesn't change anything in the generated compose files.

**Comment by cherrycl** [comment]
@lenny-intel Smoke Test failed because of the incorrect SHA. Not sure if rebase the PR can help.

File: compose-builder/Makefile [hunk 1]
```diff
@@ -598,9 +676,11 @@ ifeq (no-secty, $(filter no-secty,$(ARGS)))
 else
 	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-security.yml
 	ifneq (mqtt-bus, $(filter mqtt-bus,$(ARGS)))
-		ifneq (zmq-bus, $(filter zmq-bus,$(ARGS)))
-		    # Using secure redis messagebus by default if MQTT/ZMQ bus not specified
-			COMPOSE_FILES:=$(COMPOSE_FILES) -f add-secure-redis-messagebus.yml
+		ifneq (nats-bus, $(filter nats-bus,$(ARGS)))
+			ifneq (zmq-bus, $(filter zmq-bus,$(ARGS)))
```
Comment (jim-wang-intel) [inline compose-builder/Makefile:680]: could we just use BUS variable to do the logic or it  is hard in make file?

File: compose-builder/gen_nats_messagebus_compose_ext.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# /*******************************************************************************
+#  * Copyright 2021 Intel Corporation.
```
Comment (jim-wang-intel) [inline compose-builder/gen_nats_messagebus_compose_ext.sh:3]:
```suggestion
#  * Copyright 2022 Intel Corporation.
```

File: compose-builder/add-nats-messagebus.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# /*******************************************************************************
+#  * Copyright 2021 Intel Corporation.
```
Comment (jim-wang-intel) [inline compose-builder/add-nats-messagebus.yml:2]:
```suggestion
#  * Copyright 2022 Intel Corporation.
```

File: compose-builder/add-nats-messagebus-device-template.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# /*******************************************************************************
+#  * Copyright 2021 Intel Corporation.
```
Comment (jim-wang-intel) [inline compose-builder/add-nats-messagebus-device-template.yml:2]:
```suggestion
#  * Copyright 2022 Intel Corporation.
```

File: compose-builder/add-nats-messagebus-app-template.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# /*******************************************************************************
+#  * Copyright 2021 Intel Corporation.
```
Comment (jim-wang-intel) [inline compose-builder/add-nats-messagebus-app-template.yml:2]:
```suggestion
#  * Copyright 2022 Intel Corporation.
```

**Review by jim-wang-intel — COMMENTED** [review]
copyright year 2022

File: compose-builder/Makefile (hunk 1 above)
Comment (lenny-intel) [reply compose-builder/Makefile:701]: I like it! Will try.

Comment (lenny-intel) [reply compose-builder/Makefile:701]: Works and much cleaner. THX!

Comment (jim-wang-intel) [reply compose-builder/Makefile:701]: Awesome!

**Review by AlexCuse — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**